Fix and test rasterio 'rio' CLI entry point

ceholden opened the pull request:
The `rasterio` recipe has an out-of-date entry point definition for the rasterio command line interface. When calling `rio`:

```
> rio --help
Traceback (most recent call last):
  File "/projectnb/landsat/datasets/DATACUBE/conda/envs/tilezilla/bin/rio", line 4, in <module>
    from rasterio.rio.main import cli
ImportError: cannot import name cli
```

Rasterio switched from rasterio.rio.main:main to rasterio.rio.main:main_group on [May 30, 2015](https://github.com/mapbox/rasterio/commit/f738f1cd477b1dead4d2988575f205ada869c80d). I've fixed the reference in `recipe/meta.yaml` and added a small test in `recipe/run_test.py` to ensure the rasterio CLI can at least get to the `rio --version` option.

This is my first PR to conda-forge so please let me know if I can improve anything. Thanks very much for maintaining these builds!

conda-forge-admin commented:
Hi! This is the friendly conda-forge-admin automated user.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe```) and found it was in an excellent condition.

ocefpaf commented:
@ceholden thanks for the contribution! Consider adding yourself to the [maintainers list](https://github.com/conda-forge/rasterio-feedstock/pull/3/files#diff-e178b687b10a71a3348107ae3154e44cR43).

ocefpaf commented:
Maybe the simplest way to test this is with a command test in the `meta.yaml` file.

```yaml
 test:
     imports:
         - rasterio
     commands:
         - rio --help
```

ceholden commented:
Thanks, @ocefpaf ! I completely forgot about [the test section in the meta.yaml specification](http://conda.pydata.org/docs/building/meta-yaml.html?#test-section).

I added myself to maintainer list and updated the CLI test as recommended.

ocefpaf commented:
> I added myself to maintainer list and updated the CLI test as recommended.

Thanks @ceholden! Take a look at other feedsotcks of interest: `gdal`, `fiona`, etc. We are trying to get more people into conda-forge :wink:

pelson commented:
👍  - thanks for the contribution @ceholden!
